rfcs/0021-ckb-address-format/0021-ckb-address-format.md

```diff
@@ -38,13 +38,28 @@ payload = 0x01 | code_hash_index | single_arg
 To translate payload to lock script, one can convert code_hash_index to code_hash and hash_type with the following *popular code_hash table*. And single_arg as the args.
 
 
-| code_hash_index |        code_hash     |   code_type  |      args     |
-|:---------------:|----------------------|:------------:|---------------|
-|      0x00       | SECP256K1 + blake160 |     Type     |  blake160(PK) |
-|      0x01       | SECP256K1 + hash160  |     Type     |  hash160(PK)  |
+| code_hash_index |        code_hash     |   code_type  |          args           |
+|:---------------:|----------------------|:------------:|-------------------------|
+|      0x00       | SECP256K1 + blake160 |     Type     |  blake160(PK)*          |
+|      0x01       | SECP256K1 + multisig |     Type     |  multisig script hash** |
 
 \* The blake160 here means the leading 20 bytes truncation of Blake2b hash result.
-\* The hash160 here means Bitcoin address calculation *double hash* algorithm, which means `hash160(PK) := Ripemd160(SHA256(PK))`.
+
+\*\* The *multisig script hash* is the 20 bytes blake160 hash of multisig script. The multisig script should be assembled in the following format:
+
+```
+S | R | M | N | Pubkey1 | Pubkey2 | ...
+```
+
+Where S/R/M/N are four single byte unsigned integers, ranging from 0 to 255, and Pubkeys are SECP256K1 public keys. S is format version, currently fixed to 0. M/N means the user must provide M of N signatures to unlock the cell. And R means the provided signatures at least match the first R items of the Pubkey list.
```

Comment:
What's the use of S?

Reply:
In the original design S is a reserved field for future use, occupied 1 byte.

Reply:
For what kind of use? If there's no explicit use I prefer to remove it. We can use a new `code_hash_index` if we want to add a new multisig format in the future.

Reply:
Not sure yet, it brings some extensibility, 
we may use it to store flags to enable the new features. Or use it as a version to allow compatible changes in the future.

Of cause, use `code_hash_index ` can also achieve the above. what do you think? @CipherWang

Reply:
The code https://github.com/nervosnetwork/ckb-system-scripts/pull/60 is supposed to be freeze, we decide to do not touch the current code, we still keep the `S`.